linear_mem_attention_pytorch/linear_mem_attn_torch.py

```diff
@@ -30,6 +30,7 @@ def summarize_chunk(
         attn_weights = torch.einsum("bqhd,bkhd->bqhk", query, key)
         if mask is not None:
             max_neg = -torch.finfo(attn_weights.dtype).max
+            mask = mask.type(torch.bool)
```

Comment:
better: 
`if mask.dtype != torch.bool: mask = mask.bool()`
so we can avoid a type conversion if not needed